browser/ui/tabs/split_view_browser_data.cc

```diff
@@ -34,35 +27,26 @@ SplitViewBrowserData::~SplitViewBrowserData() {
   }
 }
 
-void SplitViewBrowserData::TileTabs(const Tile& tile) {
+void SplitViewBrowserData::TileTabs(const TabTile& tile) {
   CHECK(!IsTabTiled(tile.first));
   CHECK(!IsTabTiled(tile.second));
 
-  TabStripModel* model = is_testing_ ? tab_strip_model_for_testing_.get()
-                                     : GetBrowser().tab_strip_model();
-  if (model) {
-    CHECK_LT(model->GetIndexOfTab(tile.first),
-             model->GetIndexOfTab(tile.second));
-  }
+  auto& model = tab_strip_model_adapter_.tab_strip_model();
+  CHECK_LT(model.GetIndexOfTab(tile.first), model.GetIndexOfTab(tile.second));
 
   tiles_.push_back(tile);
 
   tile_index_for_tab_[tile.first] = tiles_.size() - 1;
   tile_index_for_tab_[tile.second] = tiles_.size() - 1;
 
-  if (tab_strip_model_adapter_) {
-    bool tabs_are_adjacent = false;
-    if (model) {
-      tabs_are_adjacent =
-          tab_strip_model_adapter_->SynchronizePinnedState(tile, tile.first);
-      tabs_are_adjacent |= tab_strip_model_adapter_->SynchronizeGroupedState(
-          tile, /*source=*/tile.first,
-          model->GetTabGroupForTab(model->GetIndexOfTab(tile.first)));
-    }
+  bool tabs_are_adjacent =
+      tab_strip_model_adapter_.SynchronizePinnedState(tile, tile.first);
+  tabs_are_adjacent |= tab_strip_model_adapter_.SynchronizeGroupedState(
+      tile, /*source=*/tile.first,
+      model.GetTabGroupForTab(model.GetIndexOfTab(tile.first)));
```

Comment:
nit: maybe worth joining these together, rather than an `|=` just underneath
```suggestion
  bool tabs_are_adjacent =
      tab_strip_model_adapter_.SynchronizePinnedState(tile, tile.first)
    || tab_strip_model_adapter_.SynchronizeGroupedState(tile, /*source=*/tile.first, model.GetTabGroupForTab(model.GetIndexOfTab(tile.first)));
```

Reply:
Or do we do it this way because calling both is a requirement? Does `|=` shortcircuit if it's already true?

Reply:
So this is a bitwise operation. However, if we bring it to a single expression, by using `|` rather than `|=`, the compiler warn against it, because it suspect maybe you were tying to do `||` and not really a bitwise operation. To suppress the warning I would have to use `int`, which makes the whole code more verbose.